test/conformance/cts_exe.py

```diff
@@ -18,18 +20,24 @@
 
     parser = ArgumentParser()
     parser.add_argument("--test_command", help="Ctest test case")
-
     args = parser.parse_args()
-    result = subprocess.Popen([args.test_command, '--gtest_brief=1'], stdout = subprocess.PIPE, text = True)  # nosec B603
+    result = subprocess.Popen([args.test_command, '--gtest_brief=1'], stdout=subprocess.PIPE,
+                              stderr=subprocess.STDOUT, text=True)  # nosec B603
 
     pat = re.compile(r'\[( )*FAILED( )*\]')
-    for line in result.stdout.readlines():
+    output_list = []
+    for line in result.stdout:
+        output_list.append(line)
         if pat.search(line):
             test_case = line.split(" ")[5]
             test_case = test_case.rstrip(',')
             print(test_case)
 
-    result.communicate()
-    rc = result.returncode
+    print("#### OUTPUT ####", file=sys.stderr)
+    for output in output_list:
+        print(output, file=sys.stderr)
+    print("#### OUTPUT_END ####", file=sys.stderr)
+
+    rc = result.wait()
```

Comment:
You are printing the output that possibly is not complete and wait for the process to complete afterwards. Shouldn't the output be printed after the process is finished? If yes, then you could use `subprocess.run()` which waits for the process to complete by default.

Reply:
I've changed it